tornado/speedups.c

```diff
@@ -14,13 +17,35 @@ static PyObject* websocket_mask(PyObject* self, PyObject* args) {
         return NULL;
     }
 
+    uint32_mask = ((uint32_t*)mask)[0];
+
     result = PyBytes_FromStringAndSize(NULL, data_len);
     if (!result) {
         return NULL;
     }
     buf = PyBytes_AsString(result);
+
+    if (sizeof(size_t) >= 8) {
+        uint64_mask = uint32_mask;
+        uint64_mask = (uint64_mask << 32) | uint32_mask;
+
+        while (data_len >= 8) {
+            ((uint64_t*)buf)[0] = ((uint64_t*)data)[0] ^ uint64_mask;
```

Comment:
minor style tip, feel free to ignore: this sort of thing is more succinctly written as

```c
    *(uint64_t*)buf = *(uint64_t*)data ^ uint64_mask;
```